base/exports.jl

```diff
@@ -1180,8 +1180,10 @@ export
 # multiprocessing
     addprocs,
     asyncmap,
+    CachingPool,
     ClusterManager,
     default_worker_pool,
+    define,
```

Comment:
It is generic, but that is the beauty of multiple dispatch.
(That said it still might be too generic, I'm not sure either way)